elide-contrib/elide-dynamic-config-helpers/src/main/java/com/yahoo/elide/contrib/dynamicconfighelpers/validator/DynamicConfigValidator.java

```diff
@@ -0,0 +1,172 @@
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Locale;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@Slf4j
+@Data
+public class DynamicConfigValidator {
+
+    private static ElideTableConfig elideTableConfig;
+    private static ElideSecurityConfig elideSecurityConfig;
+    private static Map<String, Object> variables;
+    private static final String[] SQL_DISALLOWED_WORDS = new String[] { "DROP ", "TRUNCATE ", "DELETE ", "INSERT ",
+            "UPDATE " };
+    private static final String[] ROLE_NAME_DISALLOWED_WORDS = new String[] { "," };
+
+    public static void main(String[] args) {
+
+        if (args == null || args.length == 0) {
+            usage();
+            throw new IllegalStateException("No Arguments provided!");
+        }
+
+        if (args.length > 1 || DynamicConfigHelpers.isNullOrEmpty(args[0])) {
+            usage();
+            throw new IllegalStateException("Expecting One non-empty argument only!");
+        }
+
+        File file = new File(args[0]);
+        String absoluteBasePath = DynamicConfigHelpers.formatFilePath(file.getAbsolutePath());
+        log.info("Absolute Path for Model Configs Directory: " + absoluteBasePath);
+
+        if (!file.isDirectory()) {
+            usage();
+            throw new IllegalStateException("Model Configs Directory doesn't exists");
+        }
+
+        readVariableConfig(absoluteBasePath);
+        if (readSecurityConfig(absoluteBasePath)) {
+            validateRoleInSecurityConfig(elideSecurityConfig);
+        }
+        if (readTableConfig(absoluteBasePath)) {
+            validateSqlInTableConfig(elideTableConfig);
+        }
+
+        log.info("Configs Validation Passed!");
+    }
+
+    private static boolean readVariableConfig(String absoluteBasePath) {
+        boolean isVariableConfig = exists(absoluteBasePath + DynamicConfigHelpers.VARIABLE_CONFIG_PATH);
+        try {
+            variables = isVariableConfig ? DynamicConfigHelpers.getVaribalesPojo(absoluteBasePath)
+                    : Collections.<String, Object>emptyMap();
+        } catch (Exception e) {
+            throw new IllegalStateException("Error while parsing variable config at location: " + absoluteBasePath, e);
+        }
+        return isVariableConfig;
+    }
+
+    private static boolean readSecurityConfig(String absoluteBasePath) {
+        boolean isSecurityConfig = exists(absoluteBasePath + DynamicConfigHelpers.SECURITY_CONFIG_PATH);
+        if (isSecurityConfig) {
+            String securityConfigContent = DynamicConfigHelpers
+                    .readConfigFile(new File(absoluteBasePath + DynamicConfigHelpers.SECURITY_CONFIG_PATH));
+            validateConfigForMissingVariables(securityConfigContent, variables);
+            try {
+                elideSecurityConfig = DynamicConfigHelpers.getElideSecurityPojo(absoluteBasePath, variables);
+            } catch (Exception e) {
```

Comment:
As a general rule, most code should not throw or catch Exception.  Please be explicit.

Reply:
Only checked exceptions need to be mapped to IllegalStateException.  Runtime exceptions don't need to be swallowed here.

Reply:
updated